[ARM/Linux] Build failure on ARM (softfp)

Cross build on ARM (softp) failed with the following message:
```
[  7%] Building C object src/pal/src/CMakeFiles/coreclrpal.dir/safecrt/xtow_s.c.o
cd /home/parjong/projects/toolchains/coreclr/bin/obj/Linux.tm1.Debug/src/pal/src && /usr/bin/clang-3.6  -DARM -DBIT32=1 -DBUILDENV_CHECKED=1 -DCORECLR=1 -DDEBUG -DDISABLE_CONTRACTS -DFEATURE_PAL=1 -DLINUX32 -DLP64
COMPATIBLE=1 -DPIC=1 -DPLATFORM_UNIX=1 -DURTBLDENV_FRIENDLY=Checked -D_ARM_ -D_DBG -D_DEBUG -D_FILE_OFFSET_BITS=64 -Wall -std=c11 -g -O0 -I/home/parjong/projects/toolchains/coreclr/bin/obj/Linux.tm1.Debug/src/pal/
src -I/home/parjong/projects/toolchains/coreclr/src/pal/src -I/home/parjong/projects/toolchains/coreclr/src/pal/inc -I/home/parjong/projects/toolchains/coreclr/src/pal/../inc -isystem /usr/local/include -I/home/pa
rjong/projects/toolchains/coreclr/src/pal/src/include    -target armv7-linux-gnueabi -mthumb -mfpu=vfpv3 --sysroot=/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.armv7l.0 -I/hom
e/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.armv7l.0/usr/lib/gcc/armv7l-tizen-linux-gnueabi/4.9.2/include/c++/ -I/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BU
ILD-ROOTS/scratch.armv7l.0/usr/lib/gcc/armv7l-tizen-linux-gnueabi/4.9.2/include/c++/armv7l-tizen-linux-gnueabi/ -I/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.armv7l.0/usr/inc
lude/llvm/ -I/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.armv7l.0/usr/include/llvm-c/ -I/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.arm
v7l.0/usr/local/include -I/home/parjong/projects/toolchains/coreclr/cross/rootfs/tm1-new/BUILD-ROOTS/scratch.armv7l.0/usr/include -Wno-deprecated-declarations -fno-omit-frame-pointer -fms-extensions -fwrapv -fstac
k-protector-strong -ferror-limit=4096 -Werror -Wno-unused-private-field -Wno-unused-variable -Wno-microsoft -Wno-tautological-compare -Wno-constant-logical-operand -Wno-unknown-warning-option -Wno-invalid-offsetof
 -Wno-incompatible-ms-struct -fsigned-char -mfloat-abi=softfp -fexceptions -fPIC -o CMakeFiles/coreclrpal.dir/safecrt/xtow_s.c.o   -c /home/parjong/projects/toolchains/coreclr/src/pal/src/safecrt/xtow_s.c
In file included from /home/parjong/projects/toolchains/coreclr/src/vm/appdomain.cpp:6:
In file included from /home/parjong/projects/toolchains/coreclr/src/vm/common.h:380:
/home/parjong/projects/toolchains/coreclr/src/vm/callingconvention.h:1099:25: error: no member named 'IsHFA' in 'TypeHandle'
        if (thValueType.IsHFA())
            ~~~~~~~~~~~ ^
```
It seems that the code at callingconvention.h:1099 should be enabled only if FEATURE_HFA is enabled.

--- a/src/vm/callingconvention.h
+++ b/src/vm/callingconvention.h
@@ -1094,10 +1094,12 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
         // the interop "native value types".
         fRequiresAlign64Bit = thValueType.RequiresAlign8();
 
+#ifdef FEATURE_HFA
         // Handle HFAs: packed structures of 1-4 floats or doubles that are passed in FP argument
         // registers if possible.
         if (thValueType.IsHFA())
             fFloatingPoint = true;
+#endif
 
         break;
     }

